src/ripple/overlay/impl/Message.cpp

```diff
@@ -109,22 +109,41 @@ Message::compress()
 }
 
 /** Set payload header
- * Uncompressed message header
- * 47-42    Set to 0
- * 41-16    Payload size
- * 15-0	    Message Type
- * Compressed message header
- * 79       Set to 0, indicates the message is compressed
- * 78-76    Compression algorithm, value 1-7. Set to 1 to indicate LZ4
- * compression 75-74    Set to 0 73-48    Payload size 47-32	Message Type
- * 31-0     Uncompressed message size
- */
+
+    The header is a variable-sized structure that contains information about
+    the type of the message and the length and encoding of the payload.
+
+    The first 6 bits determine whether a message is compressed or uncompressed
+    and determine the type of compression:
+
+    All multi-byte values are represented in big endian.
+
+    For uncompressed messages (6 bytes), numbering bits from left to right:
+
+        - The first 6 bits are set to 0.
+        - The next 26 bits represent the payload size.
+        - The remaining 16 bits represent the message type.
+
+    For compressed messages (10 bytes), numbering bits from left to right:
+
+        - The first 32 bits, together, represent the compression algorithm
+          and payload size:
+            - The first bit is set to 1 to indicate the message is compressed.
+            - The next 3 bits indicate the compression algorithm.
+            - The remaining 28 bits represent the payload size.
+        - The next 16 bits represent the message type.
+        - The remaining 32 bits are the uncompressed message size.
+
+    @note While nominally a part of the wire protocol, the framing is subject
+          to change; future versions of the code may negotiate the use of
+          substantially different framing.
+*/
 void
 Message::setHeader(
     std::uint8_t* in,
     std::uint32_t payloadBytes,
     int type,
-    Algorithm comprAlgorithm,
+    Algorithm compression,
     std::uint32_t uncompressedBytes)
 {
     auto h = in;
```

Comment:
Should we set 6 bits to 0 in `pack`? And the comment should be updated because we use `pack` to pack the payload size and the uncompressed message size. The comment doesn't apply in the latter case.

Reply:
We can't do so willy-nilly: it may truncate a message. Of course, any such message would be over 64MB, so we should never send it, but that's a larger change that I'd like, at this point.